tests/integration/devfile/cmd_devfile_storage_test.go

```diff
@@ -26,373 +26,374 @@ var _ = Describe("odo devfile storage command tests", func() {
-			helper.Cmd("odo", args...).ShouldPass()
+				Expect(found).To(BeTrue())
+				Expect(len(output)).To(Equal(2))
 
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+				helper.Cmd("odo", "delete", "-f", "--context", commonVar.Context).ShouldPass()
 
-			storageName := helper.RandString(5)
-
-			helper.Cmd("odo", "storage", "create", storageName, "--path", "/data", "--context", commonVar.Context).ShouldPass()
-
-			args = []string{"push", "--context", commonVar.Context}
-			helper.Cmd("odo", args...).ShouldPass()
+				// check if the owner reference is set on the source code PVC properly or not
+				commonVar.CliRunner.WaitAndCheckForTerminatingState("pvc", commonVar.Project, 1)
+			})
+		})
 
-			// Verify the pvc size
-			storageSize := commonVar.CliRunner.GetPVCSize(cmpName, storageName, commonVar.Project)
-			Expect(storageSize).To(ContainSubstring("1Gi"))
+		When("devfile storage create command is executed and doing odo push", func() {
+			BeforeEach(func() {
+				helper.Cmd("odo", "storage", "create", storageNames[0], "--path", pathNames[0], "--size", sizes[0], "--context", commonVar.Context).ShouldPass()
+			})
+
+			It("should error if same path is provided again", func() {
+				helper.Cmd("odo", "storage", "create", storageNames[1], "--path", pathNames[0], "--size", sizes[1], "--context", commonVar.Context).ShouldFail()
+			})
+
+			It("should error if same storage name is provided again", func() {
+				helper.Cmd("odo", "storage", "create", storageNames[0], "--path", pathNames[1], "--size", sizes[1], "--context", commonVar.Context).ShouldFail()
+			})
+
+			It("should list output in json format", func() {
+				actualStorageList := helper.Cmd("odo", "storage", "list", "--context", commonVar.Context, "-o", "json").ShouldPass().Out()
+				valuesSL := gjson.GetMany(actualStorageList, "kind", "items.0.kind", "items.0.metadata.name", "items.0.spec.size", "items.0.spec.path", "items.0.spec.containerName", "items.0.status")
+				expectedSL := []string{"List", "Storage", storageNames[0], sizes[0], pathNames[0], "runtime", "Not Pushed"}
+				Expect(helper.GjsonMatcher(valuesSL, expectedSL)).To(Equal(true))
+			})
+
+			It("should list storage in not pushed state", func() {
+				stdOut := helper.Cmd("odo", "storage", "list", "--context", commonVar.Context).ShouldPass().Out()
+				helper.MatchAllInOutput(stdOut, []string{storageNames[0], pathNames[0], sizes[0], "Not Pushed", cmpName})
+				helper.DontMatchAllInOutput(stdOut, []string{"CONTAINER", "runtime"})
+			})
+
+			When("odo push is executed", func() {
+				var stdOut string
+
+				BeforeEach(func() {
+					helper.Cmd("odo", "push", "--context", commonVar.Context).ShouldPass()
+					stdOut = helper.Cmd("odo", "storage", "list", "--context", commonVar.Context).ShouldPass().Out()
```

Comment:
If possible, please place this line in the next `It` block, where the value of `stdOut` is used (and you could remove the global `stdOut` variable) - same in different places

Reply:
Understood!